It does not build in offline mode:

```
[    7s] /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Game/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Game/OpenRA.Game.csproj]
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Game/OpenRA.Game.csproj" (default targets) -- FAILED.
[    7s] Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (1) is building "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Cnc/OpenRA.Mods.Cnc.csproj" (3) on node 1 (default targets).
[    7s] PrepareForBuild:
[    7s]   Creating directory "obj/Debug/".
[    7s] /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Cnc/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Cnc/OpenRA.Mods.Cnc.csproj]
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Cnc/OpenRA.Mods.Cnc.csproj" (default targets) -- FAILED.
[    7s] Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (1) is building "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Utility/OpenRA.Utility.csproj" (4) on node 1 (default targets).
[    7s] /usr/lib/mono/msbuild/Current/bin/Microsoft.Common.CurrentVersion.targets(1193,5): error MSB3644: The reference assemblies for .NETFramework,Version=v5.0 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Utility/OpenRA.Utility.csproj]
[    7s] _CleanRecordFileWrites:
[    7s]   Creating directory "obj/Debug/".
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Utility/OpenRA.Utility.csproj" (default targets) -- FAILED.
[    7s] Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (1) is building "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Server/OpenRA.Server.csproj" (5) on node 1 (default targets).
[    7s] /usr/lib/mono/msbuild/Current/bin/Microsoft.Common.CurrentVersion.targets(1193,5): error MSB3644: The reference assemblies for .NETFramework,Version=v5.0 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Server/OpenRA.Server.csproj]
[    7s] _CleanRecordFileWrites:
[    7s]   Creating directory "obj/Debug/".
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Server/OpenRA.Server.csproj" (default targets) -- FAILED.
[    7s] Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (1) is building "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.D2k/OpenRA.Mods.D2k.csproj" (6) on node 1 (default targets).
[    7s] PrepareForBuild:
[    7s]   Creating directory "obj/Debug/".
[    7s] /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.D2k/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.D2k/OpenRA.Mods.D2k.csproj]
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.D2k/OpenRA.Mods.D2k.csproj" (default targets) -- FAILED.
[    7s] Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (1) is building "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Platforms.Default/OpenRA.Platforms.Default.csproj" (7) on node 1 (default targets).
[    7s] PrepareForBuild:
[    7s]   Creating directory "obj/Debug/".
[    7s] /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Platforms.Default/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Platforms.Default/OpenRA.Platforms.Default.csproj]
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Platforms.Default/OpenRA.Platforms.Default.csproj" (default targets) -- FAILED.
[    7s] Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (1) is building "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Common/OpenRA.Mods.Common.csproj" (8) on node 1 (default targets).
[    7s] PrepareForBuild:
[    7s]   Creating directory "obj/Debug/".
[    7s] /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Common/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Common/OpenRA.Mods.Common.csproj" (default targets) -- FAILED.
[    7s] Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (1) is building "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Launcher/OpenRA.Launcher.csproj" (9) on node 1 (default targets).
[    7s] /usr/lib/mono/msbuild/Current/bin/Microsoft.Common.CurrentVersion.targets(1193,5): error MSB3644: The reference assemblies for .NETFramework,Version=v5.0 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Launcher/OpenRA.Launcher.csproj]
[    7s] _CleanRecordFileWrites:
[    7s]   Creating directory "obj/Debug/".
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Launcher/OpenRA.Launcher.csproj" (default targets) -- FAILED.
[    7s] Done Building Project "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target(s)) -- FAILED.
[    7s] 
[    7s] Build FAILED.
[    7s] 
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target) (1) ->
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Game/OpenRA.Game.csproj" (default target) (2) ->
[    7s] (ResolvePackageAssets target) -> 
[    7s]   /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Game/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Game/OpenRA.Game.csproj]
[    7s] 
[    7s] 
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target) (1) ->
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Cnc/OpenRA.Mods.Cnc.csproj" (default target) (3) ->
[    7s]   /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Cnc/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Cnc/OpenRA.Mods.Cnc.csproj]
[    7s] 
[    7s] 
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target) (1) ->
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Utility/OpenRA.Utility.csproj" (default target) (4) ->
[    7s] (GetReferenceAssemblyPaths target) -> 
[    7s]   /usr/lib/mono/msbuild/Current/bin/Microsoft.Common.CurrentVersion.targets(1193,5): error MSB3644: The reference assemblies for .NETFramework,Version=v5.0 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Utility/OpenRA.Utility.csproj]
[    7s] 
[    7s] 
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target) (1) ->
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Server/OpenRA.Server.csproj" (default target) (5) ->
[    7s]   /usr/lib/mono/msbuild/Current/bin/Microsoft.Common.CurrentVersion.targets(1193,5): error MSB3644: The reference assemblies for .NETFramework,Version=v5.0 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Server/OpenRA.Server.csproj]
[    7s] 
[    7s] 
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target) (1) ->
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.D2k/OpenRA.Mods.D2k.csproj" (default target) (6) ->
[    7s] (ResolvePackageAssets target) -> 
[    7s]   /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.D2k/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.D2k/OpenRA.Mods.D2k.csproj]
[    7s] 
[    7s] 
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target) (1) ->
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Platforms.Default/OpenRA.Platforms.Default.csproj" (default target) (7) ->
[    7s]   /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Platforms.Default/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Platforms.Default/OpenRA.Platforms.Default.csproj]
[    7s] 
[    7s] 
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target) (1) ->
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Common/OpenRA.Mods.Common.csproj" (default target) (8) ->
[    7s]   /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1004: Assets file '/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Common/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]
[    7s] 
[    7s] 
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.sln" (Build target) (1) ->
[    7s] "/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Launcher/OpenRA.Launcher.csproj" (default target) (9) ->
[    7s] (GetReferenceAssemblyPaths target) -> 
[    7s]   /usr/lib/mono/msbuild/Current/bin/Microsoft.Common.CurrentVersion.targets(1193,5): error MSB3644: The reference assemblies for .NETFramework,Version=v5.0 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/home/abuild/rpmbuild/BUILD/openra-20210103/OpenRA.Launcher/OpenRA.Launcher.csproj]
[    7s] 
[    7s]     0 Warning(s)
[    7s]     8 Error(s)
[    7s] 
[    7s] Time Elapsed 00:00:00.88
[    7s] error: Bad exit status from /var/tmp/rpm-tmp.4H6Xnm (%build)
```

You only packaged the nuget packages, not the cache: https://build.opensuse.org/source/games:openra/release/nuget-cache.tar.gz

--- a/.github/workflows/packaging.yml
+++ b/.github/workflows/packaging.yml
@@ -8,6 +8,30 @@ on:
     - 'devtest-*'
 
 jobs:
+  source:
+    name: Source Tarball
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Clone Repository
+        uses: actions/checkout@v2
+
+      - name: Prepare Environment
+        run: echo "GIT_TAG=${GITHUB_REF#refs/tags/}" >> ${GITHUB_ENV}
+
+      - name: Package Source
+        run: |
+          mkdir -p build/source
+          ./packaging/source/buildpackage.sh "${GIT_TAG}" "${PWD}/build/source"
+
+      - name: Upload Packages
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          tag: ${{ github.ref }}
+          overwrite: true
+          file_glob: true
+          file: build/source/*
+
   linux:
     name: Linux AppImages
     runs-on: ubuntu-20.04

